services/checks/funcs_test.go

```diff
@@ -96,3 +96,112 @@ Traceback (most recent call last):
 	}}
 	assert.Equal(t, expected, res)
 }
+
+func TestAdditionalContext(t *testing.T) {
+	funcs, err := GetFuncsForVersion(1)
+	require.NoError(t, err)
+	contextFuncs := GetAdditionalContext()
+
+	testCases := []struct {
+		name   string
+		script string
+		err    string
+		result []check.Result
+	}{
+		{
+			name: "too many args",
+			script: strings.TrimSpace(`
+def check_context(rows, context):
+    func = context.get("ip_is_private", fail)
+    print("result =", func(1, 2))
+
+    return [{
+        "summary": "IP Address Check",
+        "severity": "warning",
+    }]
+	`),
+			err: strings.TrimSpace(`
+thread too many args: failed to execute function check_context: ip_is_private: expected 1 argument, got 2
+Traceback (most recent call last):
+  TestAdditionalContext/too_many_args:3:27: in check_context
+  <builtin>: in ip_is_private
+	`) + "\n",
+			result: nil,
+		},
+		{
+			name: "invalid arg",
+			script: strings.TrimSpace(`
+def check_context(rows, context):
+    func = context.get("ip_is_private", fail)
+    print("result =", func(1))
```

Comment:
How about `print("result =", func("not an IP address"))` ?

Reply:
no that would be a string argument and this test case is to check for the case in which we get a wrong type of argument, if I use string here too it reduces the coverage and defeats the purpose of this test case

Reply:
Yes, I mean how about another case, when type is correct (strinng) but content is wrong)